addons/main/functions/fn_createTable.sqf

```diff
@@ -79,7 +81,7 @@ private _dummy = "Land_HelipadEmpty_F" createVehicleLocal _markerPos;
 _markerPos set [2, (0 max getTerrainHeightASL _markerPos) + 1];
 _dummy enableSimulation false;
 _dummy setPosASL _markerPos;
-_dummy setDir _markerDir;
+_dummy setDir (_markerDir + ([0,90] select (_tableWidth > _tableLength)));
```

Comment:
I think this is where the messed up terrain normals is coming from. You should set the dir of the MARKER earlier on and work around that, as the dummy is not used at all for getting the correct terrain normal.

OR:

line 144: 
`_normal = [_normal, _tableDir -_markerDir, 2] call BIS_fnc_rotateVector3D`

Perhaps swapping _markerDir to _dummyDir would work.